File: app/src/main/java/com/konektedi/vs/Utilities/Api/ApiUtilities.java
package com.konektedi.vs.Utilities.Api;

/**
 * Created by Sy on b/9/2018.
 */

public class ApiUtilities {

    public static final String BASE_URL = "http://192.168.43.188/konektedi_vs/api/v1/";
//    public static final String BASE_URL = "http://vs.konektedi.com/api/v1/";

    public static Api getClient() {
        return ApiClient.getClient(BASE_URL).create(Api.class);
    }

}

File: app/src/main/java/com/konektedi/vs/utilities/api/ApiUtilities.kt
package com.konektedi.vs.utilities.api

import com.google.gson.Gson
import com.konektedi.vs.utilities.models.ApiError
import okhttp3.ResponseBody
import retrofit2.Response

/*
    * Returns all Api error messages as Strings
 */
fun getError(response: Response<ResponseBody>): String? {
    val gson = Gson()
    val message = gson.fromJson(response.errorBody()?.charStream(), ApiError::class.java)
    return message.message
}

/*
    * Returns a certain value from server requests
 */
fun getValue(response: Response<ResponseBody>): Int? {
    val gson = Gson()
    val value = gson.fromJson(response.body()?.charStream(), ApiError::class.java)
    return value.value
}